src/index.ts

```diff
@@ -65,26 +65,138 @@ export default class TypedEnv<S extends EnvSchema> extends Error {
     );
   }
 
-  configEnvironment(filePath = '.env') {
+  private inferValueType(value: string): 'string' | 'number' | 'boolean' {
+    // If the value is quoted, treat it as a string
+    if (this.isQuoted(value)) {
+      return 'string';
+    }
+
+    // Check if it's a boolean value (unquoted)
+    if (this.isBooleanValue(value)) {
+      return 'boolean';
+    }
+
+    // Check if it's a number (unquoted)
+    if (this.isNumericValue(value)) {
+      return 'number';
+    }
+
+    // Default to string
+    return 'string';
+  }
+
+  private isBooleanValue(value: string): boolean {
+    const trimmedValue = value.trim();
+    const lowerValue = trimmedValue.toLowerCase();
+    return ['true', 'false'].includes(lowerValue);
+  }
+
+  private isNumericValue(value: string): boolean {
+    // Empty string is not a number
+    if (value.trim() === '') {
+      return false;
+    }
+
+    // Check if it's a valid number
+    const numValue = Number(value);
+    return !isNaN(numValue) && isFinite(numValue);
+  }
+
+  public inferSchemaFromEnv(filePath = '.env'): EnvSchema {
+    // Use the optimized configEnvironment method with schema inference
+    return this.configEnvironment(filePath, true) || {};
+  }
+
+  private inferValueTypeWithQuoteInfo(
+    value: string,
+    wasQuoted: boolean,
+  ): 'string' | 'number' | 'boolean' {
+    // If the value was quoted, treat it as a string
+    if (wasQuoted) {
+      return 'string';
+    }
+
+    // Check if it's a boolean value (unquoted)
+    if (this.isBooleanValue(value)) {
+      return 'boolean';
+    }
+
+    // Check if it's a number (unquoted)
+    if (this.isNumericValue(value)) {
+      return 'number';
+    }
+
+    // Default to string
+    return 'string';
+  }
+
+  public initFromEnv(filePath = '.env'): InferSchema<S> {
```

Comment:
is this required now?

Reply:
We actually do not want an exposed function to inferSchema, non at all.
The idea is to be simply clever, which is if the user doesn't provide a schema, do not fail and rather infer schema from the .env file by default.